### PR TITLE
Add more assertions to ExtensibleEvents tests

### PR DESCRIPTION
**Pull request by andybalaam**
Being a little more explicit about what the contents of events will end up being.

### CODE REVIEW

File: test/ExtensibleEvents.test.ts [hunk 1]
```diff
@@ -49,9 +49,11 @@ describe('ExtensibleEvents', () => {
                     [M_TEXT.name]: "Hello World",
                 },
             };
-            const event = (new ExtensibleEvents()).parse(input);
+            const event: ExtensibleEvent = (new ExtensibleEvents()).parse(input);
             expect(event).toBeDefined();
             expect(event instanceof MessageEvent).toBe(true);
+            expect(event.serialize().type).toBe("m.room.message");
+            expect(event.serialize().content).toHaveProperty("body", "Hello World");
```
Comment (turt2live) [inline test/ExtensibleEvents.test.ts:56]: This sort of testing is covered by other tests?

File: test/ExtensibleEvents.test.ts [hunk 2]
```diff
@@ -104,6 +114,8 @@ describe('ExtensibleEvents', () => {
             event = extev.parse(input);
             expect(event).toBeDefined();
             expect(event instanceof MyCustomEvent).toBe(true);
+            expect(event.serialize().type).toBe("myNamespace.name");
+            expect(event.serialize().content).toHaveProperty("hello", "world");
```
Comment (turt2live) [inline test/ExtensibleEvents.test.ts:118]: not really sure what these and below are testing - `serialize()` is ultimately a function of our mock ExtensibleEvent, so it feels like we're testing that we wrote the mock correctly. It feels a bit more important to ensure that the code isn't calling serialize needlessly, so having the throw.

**Review by turt2live — COMMENTED** [review]
I'm somewhat concerned that these aren't quite testing what you're expecting them to, but certainly eligible for inclusion. 

The project is still pretty early, but the testing strategy is to cover all the functional possibilities over a range of tests rather than explicitness through each and every test - if some other test covers the output then another test doesn't have to.

**Comment by andybalaam** [comment]
No problem if you want to close this PR - it was just my first reaction reading the code, that some of the test are of the style "will fail if I didn't implement it", but not quite "demonstrates how to use the code", which can be a helpful style.

File: test/ExtensibleEvents.test.ts (hunk 2 above)
Comment (turt2live) [reply test/ExtensibleEvents.test.ts:118]: an `expect` in the constructor would probably make sense, yea. Also would apply to the `MockEvent` class.

as a minor nitpick in this area: `org.example` namespace please, like `org.example.custom` :)

File: test/ExtensibleEvents.test.ts (hunk 1 above)
Comment (turt2live) [reply test/ExtensibleEvents.test.ts:56]: ah, this isn't trying to test that it serializes back into an appropriate shape, just that the parser produces a sensible event object. The actual parsing is tested by `MessageEvent`'s tests.

File: test/ExtensibleEvents.test.ts (hunk 2 above)
Comment (andybalaam) [reply test/ExtensibleEvents.test.ts:118]: Do you mean `myNamespace`?  I'm 99% sure I copied that from you!

Comment (turt2live) [reply test/ExtensibleEvents.test.ts:118]: `myNamespace` is a variable which maps to an unstable-only value: `const myNamespace = new UnstableValue(null, "org.example.custom.event");` ;)

File: test/ExtensibleEvents.test.ts (hunk 1 above)
Comment (andybalaam) [reply test/ExtensibleEvents.test.ts:56]: Replaced with a cast to MessageEvent and then checking it actually contains `Hello World`

File: test/ExtensibleEvents.test.ts (hunk 2 above)
Comment (andybalaam) [reply test/ExtensibleEvents.test.ts]: Ah sorry, really dumb!

**Comment by andybalaam** [comment]
OK, I've adjusted the assertions to avoid serializing, but maintain the checking that the contents of the event actually get through.  I think there is some value in that, but feel free to drop it if you think it clutters the code.

**Review by turt2live — APPROVED** [review]
Thanks!

**Pull request merged**